cmd/habitat/api/response.go

```diff
@@ -17,3 +21,31 @@ func WriteResponse(w http.ResponseWriter, res interface{}) {
 	}
 	w.Write(body)
 }
+
+func WriteWebsocketError(conn *websocket.Conn, err error, res ctl.WebsocketMessage) error {
+	res.SetError(err)
+
+	err = conn.WriteJSON(res)
+	if err != nil {
+		return err
+	}
+
+	err = conn.WriteMessage(websocket.CloseMessage, []byte{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func WriteWebsocketClose(conn *websocket.Conn) {
```

Comment:
does gorilla not give functions like `ws.Close()` to do this for us?

Reply:
Yeah unfortunately conn.Close does not send a close message, it just closes the network connection. Thanks for pointing this out tho, because looking at the documentation more, they have a way of setting a close handler, which would be cleaner than doing this